### DISCUSSION TITLE
A theoretical question in RadialReturnStressUpdate

### DISCUSSION THREAD

**Discussion by xuxiaobei1995** [idaholab/moose#25353] [Q&A Modules: Solid mechanics] [answered]
Hi all,

Recently I read the code in RadialReturnStressUpdate and got confused about a theoretical question. In RadialReturnStressUpdate the final stress is computed as: 

```
  if (_apply_strain)
  {
    strain_increment -= inelastic_strain_increment;
    updateEffectiveInelasticStrain(_effective_inelastic_strain_increment);

    // Use the old elastic strain here because we require tensors used by this class
    // to be isotropic and this method natively allows for changing in time
    // elasticity tensors
    stress_new = elasticity_tensor * (strain_increment + elastic_strain_old);
  }
```

According to my understanding about the code, It seems to show when the material is in a inleastic-deformed status, the stress is only generated by the elastic strain. Or in formula it writes:
σ = C * (ε_total - ε_ine - ε_eigen) = C * ε_e

Does anyone know whether my understanding right or not? Thanks!

Xiaobei

**Comment by GiudGiud** [comment]
@dewenyushu @dschwen

**Reply by hugary1995** [reply, marked as answer]
Your understanding is correct.

**Reply by xuxiaobei1995** [reply]
Thank you hugary!
